python/ray/_private/function_manager.py

```diff
@@ -219,21 +231,30 @@ def f(*args, **kwargs):
 
     def get_execution_info(self, job_id, function_descriptor):
         """Get the FunctionExecutionInfo of a remote function.
-
         Args:
             job_id: ID of the job that the function belongs to.
             function_descriptor: The FunctionDescriptor of the function to get.
-
         Returns:
             A FunctionExecutionInfo object.
         """
         if self._worker.load_code_from_local:
             # Load function from local code.
             # Currently, we don't support isolating code by jobs,
             # thus always set job ID to NIL here.
-            job_id = ray.JobID.nil()
-            if not function_descriptor.is_actor_method():
-                self._load_function_from_local(job_id, function_descriptor)
+            function_id = function_descriptor.function_id
```

Comment:
This would be clearer
```
if already in _function_execution_info:
   return info
if load_code_from_local:
   if not actor method:
       info = self._load_function_from_local()
       if info is not None:
           return info
# Load from GCS
...
   
```